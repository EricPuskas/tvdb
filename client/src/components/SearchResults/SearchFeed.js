import React, { memo } from "react";
import ShowItem from "../Shows/ShowItem";
import uuid from "uuid/v4";
import Loader from "../Loader/Loader";

const SearchFeed = ({ search_results, error, displaySearchText, loading }) => {
  const results_count = [];
  const results = search_results.map(show => {
    if (show.overview) {
      results_count.push(show.id);
      return <ShowItem key={uuid()} show={show} />;
    } else return null;
  });

  return (
    <>
      {results_count.length > 0 ? (
        <p className="text-center">
          Found {results_count.length}{" "}
          {results_count.length === 1 ? "result" : "results"} matching "
          <strong>{displaySearchText}</strong>".
        </p>
      ) : null}
      {loading && <Loader />}
      {!error && !loading && (
        <div className="SearchResults-results">{results}</div>
      )}
      {error && <h1 className="text-center"> {error} </h1>}
    </>
  );
};

export default memo(SearchFeed);
