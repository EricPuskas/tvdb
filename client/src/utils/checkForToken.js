import jwt_decode from "jwt-decode";

// Redux Store
import store from "../store";

// Utilities
import setToken from "./setToken";

// Actions
import { getToken } from "../actions/token";

const checkForToken = () => {
  // Check for token
  if (!localStorage.token) {
    store.dispatch(getToken());
    setToken(localStorage.token);
  } else {
    setToken(localStorage.token);
    // Decode token to be able to check if it expired
    const decoded = jwt_decode(localStorage.token);
    // Check for expired token
    const currentTime = Date.now() / 1000;
    if (decoded.exp < currentTime) {
      store.dispatch(getToken());
    }
  }
};

export default checkForToken;
