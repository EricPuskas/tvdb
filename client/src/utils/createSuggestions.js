const filterInput = input => {
  return input.replace(/[^\w\s]/gi, "");
};

const createSuggestions = (word, data) => {
  let filter = filterInput(word);
  const re = new RegExp(`${filter.toLowerCase()}.*\\B`, "g");
  return data.filter(item => re.test(item.toLowerCase()));
};

export default createSuggestions;
