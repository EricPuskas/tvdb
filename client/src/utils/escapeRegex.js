const escapeRegex = text => {
  if (text !== "" || text !== null || text !== undefined) {
    // replace all white space with +, globally (replace all occurrences)
    return text.replace(/[\s]/g, "+");
  }
};

export default escapeRegex;
