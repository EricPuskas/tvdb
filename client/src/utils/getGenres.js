import React from "react";
import uuid from "uuid/v4";

const getGenres = genres => {
  if (!genres) return null;
  return genres.map(genre => (
    <li key={uuid()}>
      <span className="tag">{genre}</span>
    </li>
  ));
};

export default getGenres;
