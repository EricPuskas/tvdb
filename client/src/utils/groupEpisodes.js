const groupEpisodes = data => {
  let seasons = [];
  for (let i = 0; i < data.length; i++) {
    if (!seasons[data[i].airedSeason]) seasons[data[i].airedSeason] = [];
    seasons[data[i].airedSeason].push({
      id: data[i].id,
      season: data[i].airedSeason,
      episode: data[i].airedEpisodeNumber,
      title: data[i].episodeName
    });
  }
  return seasons;
};

export default groupEpisodes;
