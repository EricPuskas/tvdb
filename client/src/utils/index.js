import { applyMiddleware, createStore } from "redux";
import rootReducer from "../reducers";
import { middleware } from "../store";

export const findByAttribute = (component, attr) =>
  component.find(`[data-test="${attr}"]`);

export const testStore = initState => {
  const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);
  return createStoreWithMiddleware(rootReducer, initState);
};
