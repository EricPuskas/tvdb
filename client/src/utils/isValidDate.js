import moment from "moment";

const isValidDate = date => {
  return moment(date).isValid();
};

export default isValidDate;
