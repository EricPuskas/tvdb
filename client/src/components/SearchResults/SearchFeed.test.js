import React from "react";
import { mount } from "enzyme";
import { findByAttribute } from "../../utils";
import SearchFeed from "./SearchFeed";
import { BrowserRouter } from "react-router-dom";

describe("SearchFeed Component", () => {
  describe("with Props", () => {
    const props = {
      search_results: [
        { id: 1, title: "ABC", overview: "ABCD" },
        { id: 2, title: "ABC", overview: "ABCD" },
        { id: 3, title: "ABC", overview: "ABCD" }
      ],
      error: null,
      displaySearchText: "ABC",
      loading: false
    };

    const component = mount(
      <BrowserRouter>
        <SearchFeed {...props} />
      </BrowserRouter>
    );
    const wrapper = findByAttribute(component, "search-feed");

    it("Should pass snapshot test", () => {
      expect(component.debug()).toMatchSnapshot();
    });

    it("Should render without errors", () => {
      expect(wrapper.length).toBe(1);
    });
  });
  describe("Missing Overview Prop", () => {
    const props = {
      search_results: [
        { id: 1, title: "ABC" },
        { id: 2, title: "ABC" },
        { id: 3, title: "ABC" }
      ],
      error: null,
      displaySearchText: "ABC",
      loading: false
    };

    const component = mount(
      <BrowserRouter>
        <SearchFeed {...props} />
      </BrowserRouter>
    );
    const wrapper = findByAttribute(component, "search-feed");
    const results = findByAttribute(component, "results");

    it("Should pass snapshot test", () => {
      expect(component.html()).toMatchSnapshot();
    });

    it("Should render without errors", () => {
      expect(wrapper.length).toBe(1);
    });
    it("Should display 0 results", () => {
      expect(results.length).toBe(0);
    });
  });
});
