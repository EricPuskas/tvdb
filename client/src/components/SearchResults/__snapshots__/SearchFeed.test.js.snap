// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFeed Component Missing Overview Prop Should pass snapshot test 1`] = `"<div data-test=\\"search-feed\\"><div class=\\"SearchFeed\\"></div></div>"`;

exports[`SearchFeed Component with Props Should pass snapshot test 1`] = `
"<BrowserRouter>
  <Router history={{...}}>
    <Memo(SearchFeed) search_results={{...}} error={{...}} displaySearchText=\\"ABC\\" loading={false}>
      <div data-test=\\"search-feed\\">
        <p className=\\"text-center\\" data-test=\\"results\\">
          Found 
          3
           
          results
           matching &quot;
          <strong>
            ABC
          </strong>
          &quot;.
        </p>
        <div className=\\"SearchFeed\\">
          <ShowItem show={{...}}>
            <div className=\\"ShowItem\\" data-test=\\"showitem\\">
              <Link to=\\"/shows/1\\">
                <LinkAnchor href=\\"/shows/1\\" navigate={[Function: navigate]}>
                  <a href=\\"/shows/1\\" onClick={[Function: onClick]}>
                    <div className=\\"row\\">
                      <div className=\\"col-lg-3 col-12\\">
                        <ShowHeader title={[undefined]} firstAired={[undefined]} network={[undefined]}>
                          <h2>
                             
                             
                          </h2>
                          <hr />
                          <div>
                            First Aired:
                             
                            <strong data-test=\\"firstAired\\">
                              <t format=\\"MMM DD (YYYY)\\" element={{...}} fromNow={false} toNow={false} calendar={false} ago={false} unix={false} utc={false} local={false} unit={{...}} withTitle={false} decimal={false} titleFormat=\\"\\" interval={60000} filter={[Function: filter]} onChange={[Function: onChange]}>
                                <time dateTime={{...}}>
                                  Oct 29 (2019)
                                </time>
                              </t>
                            </strong>
                          </div>
                          <div>
                            Network:
                             
                            <strong data-test=\\"network\\">
                              Unknown
                            </strong>
                          </div>
                          <hr />
                        </ShowHeader>
                      </div>
                      <div className=\\"col-lg-9 col-12\\">
                        <ShowBanner banner={[undefined]} title={[undefined]}>
                          <img data-test=\\"banner\\" onError={[Function: onError]} src=\\"no_banner.png\\" alt={[undefined]} />
                        </ShowBanner>
                      </div>
                      <div className=\\"col-12 overview\\">
                        <ShowDescription maxChars={350} overview=\\"ABCD\\">
                          <div data-test=\\"description\\">
                            <h3>
                              Overview:
                            </h3>
                            <p data-test=\\"overview\\">
                              ABCD
                            </p>
                          </div>
                        </ShowDescription>
                      </div>
                    </div>
                  </a>
                </LinkAnchor>
              </Link>
            </div>
          </ShowItem>
          <ShowItem show={{...}}>
            <div className=\\"ShowItem\\" data-test=\\"showitem\\">
              <Link to=\\"/shows/2\\">
                <LinkAnchor href=\\"/shows/2\\" navigate={[Function: navigate]}>
                  <a href=\\"/shows/2\\" onClick={[Function: onClick]}>
                    <div className=\\"row\\">
                      <div className=\\"col-lg-3 col-12\\">
                        <ShowHeader title={[undefined]} firstAired={[undefined]} network={[undefined]}>
                          <h2>
                             
                             
                          </h2>
                          <hr />
                          <div>
                            First Aired:
                             
                            <strong data-test=\\"firstAired\\">
                              <t format=\\"MMM DD (YYYY)\\" element={{...}} fromNow={false} toNow={false} calendar={false} ago={false} unix={false} utc={false} local={false} unit={{...}} withTitle={false} decimal={false} titleFormat=\\"\\" interval={60000} filter={[Function: filter]} onChange={[Function: onChange]}>
                                <time dateTime={{...}}>
                                  Oct 29 (2019)
                                </time>
                              </t>
                            </strong>
                          </div>
                          <div>
                            Network:
                             
                            <strong data-test=\\"network\\">
                              Unknown
                            </strong>
                          </div>
                          <hr />
                        </ShowHeader>
                      </div>
                      <div className=\\"col-lg-9 col-12\\">
                        <ShowBanner banner={[undefined]} title={[undefined]}>
                          <img data-test=\\"banner\\" onError={[Function: onError]} src=\\"no_banner.png\\" alt={[undefined]} />
                        </ShowBanner>
                      </div>
                      <div className=\\"col-12 overview\\">
                        <ShowDescription maxChars={350} overview=\\"ABCD\\">
                          <div data-test=\\"description\\">
                            <h3>
                              Overview:
                            </h3>
                            <p data-test=\\"overview\\">
                              ABCD
                            </p>
                          </div>
                        </ShowDescription>
                      </div>
                    </div>
                  </a>
                </LinkAnchor>
              </Link>
            </div>
          </ShowItem>
          <ShowItem show={{...}}>
            <div className=\\"ShowItem\\" data-test=\\"showitem\\">
              <Link to=\\"/shows/3\\">
                <LinkAnchor href=\\"/shows/3\\" navigate={[Function: navigate]}>
                  <a href=\\"/shows/3\\" onClick={[Function: onClick]}>
                    <div className=\\"row\\">
                      <div className=\\"col-lg-3 col-12\\">
                        <ShowHeader title={[undefined]} firstAired={[undefined]} network={[undefined]}>
                          <h2>
                             
                             
                          </h2>
                          <hr />
                          <div>
                            First Aired:
                             
                            <strong data-test=\\"firstAired\\">
                              <t format=\\"MMM DD (YYYY)\\" element={{...}} fromNow={false} toNow={false} calendar={false} ago={false} unix={false} utc={false} local={false} unit={{...}} withTitle={false} decimal={false} titleFormat=\\"\\" interval={60000} filter={[Function: filter]} onChange={[Function: onChange]}>
                                <time dateTime={{...}}>
                                  Oct 29 (2019)
                                </time>
                              </t>
                            </strong>
                          </div>
                          <div>
                            Network:
                             
                            <strong data-test=\\"network\\">
                              Unknown
                            </strong>
                          </div>
                          <hr />
                        </ShowHeader>
                      </div>
                      <div className=\\"col-lg-9 col-12\\">
                        <ShowBanner banner={[undefined]} title={[undefined]}>
                          <img data-test=\\"banner\\" onError={[Function: onError]} src=\\"no_banner.png\\" alt={[undefined]} />
                        </ShowBanner>
                      </div>
                      <div className=\\"col-12 overview\\">
                        <ShowDescription maxChars={350} overview=\\"ABCD\\">
                          <div data-test=\\"description\\">
                            <h3>
                              Overview:
                            </h3>
                            <p data-test=\\"overview\\">
                              ABCD
                            </p>
                          </div>
                        </ShowDescription>
                      </div>
                    </div>
                  </a>
                </LinkAnchor>
              </Link>
            </div>
          </ShowItem>
        </div>
      </div>
    </Memo(SearchFeed)>
  </Router>
</BrowserRouter>"
`;
